scripts/libwrapper.js

```diff
@@ -44,6 +54,7 @@ function patch_shortRest() {
        * @param {RestConfiguration} config  Configuration options for the rest.
        * @returns {boolean}                 Explicitly return `false` to prevent the rest from being started.
        */
+      if (getProperty(this, `flags.dae.rest-recovery.prevent.shortRest`)) return preventRestChatMessage("shortRest", this);

```

Comment:
I'm not sure about creating a chat message every time you click the rest buttons, I think a notification would be enough. 

You see, the average user will have just opened their character sheet and clicked on the button to rest, and if there's no feedback visible immediately, it's not going to be obvious to users that nothing is _supposed_ to happen. Since the user probably came from the actors directory, the chat tab is not going to be visible, and the user may spam click the rest button, resulting in a clogged chat.

I think this should be enough:
```js
if (getProperty(this, `flags.dae.rest-recovery.prevent.shortRest`)){
    ui.notifications.warn("Rest Recovery | " game.i18n.localize("REST-RECOVERY.Warnings.PreventedShortRest"))
    return false;
}
```
With that localization being something like `You cannot short rest because you are under an effect that prevents you from doing so!"

Reply:
Additionally, I would recommend to add an additional parameter to the `config` object that allows you to sidestep the rest prevention, so that GM's or other modules can create behavior that allows for that.

Reply:
I am going to replace ChatMessage with UI notification.

I do not know how to implement this config object extenstion and also do not kwow where to implement it.  Just started to learn/understand JS and Svelte is also brand new world to me.
I also have difficulties with code modifications, I cannot build your module into a single js file, so I modify directly under Foundry and later re-implement it in VSCODE int the repo. There should be a solution but until know I did not find it.